spec/models/subscription_spec.rb

```diff
@@ -6,7 +6,32 @@
   it { is_expected.to have_many(:notifications) }
   it { is_expected.to belong_to(:supplier) }
   it { is_expected.to validate_presence_of :supplier }
-  it { is_expected.to validate_presence_of :callback_url }
+
+  describe 'email_address or callback_url (or both) are required' do
```

Comment:
With  "(both)"  you mean that there is a case where email_address AND callback_url are required,
this might be true, but not clear from the test.

Reply:
yes, line 29 is the test for both: `when email_address and callback_url provided`